src/lib/functions/registry.js

```diff
@@ -141,6 +143,18 @@ class FunctionsRegistry {
       )
     }
 
+    // This fixes the bug described here https://github.com/netlify/zip-it-and-ship-it/issues/637
+    // If the current function's file has a '.zip' extension, we ignore it to prevent unrecoverable
+    // errors when it's called and log a descriptive message of the issue to the console.
+    if (func.mainFile.toLowerCase().endsWith(ZIP_EXTENSION)) {
+      log(
+        `${NETLIFYDEVERR} Unable to load the function ${terminalLink(chalk.yellow(name), func.url)}. Invalid file '${
```

Comment:
Rather than framing this as an error and saying that it's an invalid file, I think I would say something like "Skipped bundled function <X>. Unzip the archive to load it from source", and maybe use `NETLIFYDEVWARN` instead of `NETLIFYDEVERR`? My thinking is that this is not a fatal error, and we should try to give people instructions on how to self-serve a solution.

What do you think?

Reply:
Yes, I think yours reads better than what I had, and I agree we should treat this as a warning since it's no longer a crash scenario. Thanks 🥇